src/Wt/WTable.C

```diff
@@ -102,7 +102,11 @@ WTableRow* WTable::insertRow(int row, std::unique_ptr<WTableRow> tableRow)
   for (auto &cell : tableRow->cells_) {
     widgetAdded(cell.get());
   }
+
+  int old_end = rows_.size();
   rows_.insert(rows_.begin() + row, std::move(tableRow));
+  updateRowNumbers(std::min(old_end, row), rowCount() - 1);
```

Comment:
In what situation would this not suffice?

```suggestion
  updateRowNumbers(row, rowCount() - 1);
```